docs/design/persistent-volume-provisioning.md

```diff
@@ -0,0 +1,140 @@
+
+Static pools of storage (e.g, NFS) will have ```MinimumReplicas``` == ```MaximumReplicas```.  Creation or deletion of the resource will not occur but the controller allows recycling of the volume through the plugin.
+
+#### PersistentVolumeController API
+
+| Action | HTTP Verb | Path | Description |
+| ---- | ---- | ---- | ---- |
+| CREATE | POST | /api/{version}/persistentvolumecontrollers/ | Create instance of PersistentVolumeController|
+| GET | GET | /api/{version}persistentvolumecontrollers/{name} | Get instance of PersistentVolumeController |
+| UPDATE | PUT | /api/{version}/persistentvolumecontrollers/{name} | Update instance of PersistentVolumeController |
+| DELETE | DELETE | /api/{version}/persistentvolumecontrollers/{name} | Delete instance of PersistentVolumeController |
+| LIST | GET | /api/{version}/persistentvolumecontrollers | List instances of PersistentVolumeController|
+| WATCH | GET | /api/{version}/watch/persistentvolumecontrollers | Watch for changes to a PersistentVolumeController|
+
+
+#### PersistentVolumeControllerManager
+
+* watch PersistentVolumes of a particular type as defined by the controller's label selector.
+** PVs matching the selector but made manually count as replicas if they match the label selector
+** PVs made manually or from earlier templates may have difference size capacities.  Delete largest first.
+* create a new volume ``` if unboundReplicas < minimumReplicas && totalReplicas <= maximumReplicas``` 
+* delete a volume ``` while totalReplicas > maximumReplicas, delete largest volume ```
+
+
+#### Volumes and their plugins
+
+Use of cloud and non-cloud volumes as examples for controllers.
+
+| Volume |  Dynamic Creation | Formatting | Reclamation |
+| ---- | ---- | ---- | ---- |
+| GCE | yes via API | on mount, uses "safeFormatAndMount" | delete via API |
+| AWS | yes via API | on mount, uses "safeFormatAndMount" | delete via API |
+| OpenStack | yes via API | on mount, uses "safeFormatAndMount" | delete via API |
+| NFS | No, externally provisioned | pre-formatted during provisioning | mount volume, run "rm -rf" on volume |
+| ISCSI | No, externally provisioned | pre-formatted during provisioning | mount volume, run "rm -rf" on volume |
+| gluster  | No, externally provisioned | pre-formatted during provisioning | mount volume, run "rm -rf" on volume | ? | ? | ? |
+| ceph  | No, externally provisioned | pre-formatted during provisioning | mount volume, run "rm -rf" on volume | ? | ? | ? |
+
+#### Recycler implementation 
```

Comment:
See earlier comment about being confused, I thought we already support volume recycling.